src/lib/layer.js

```diff
@@ -409,6 +413,17 @@ export default class Layer {
 
   // Calculates uniforms
   drawLayer({uniforms = {}}) {
+    const {gl} = this.context;
+
+    // Apply polygon offset to avoid z-fighting
+    if (this.props.getPolygonOffset) {
+      gl.enable(GL.POLYGON_OFFSET_FILL);
```

Comment:
A minor concern is that layer that uses this will now "hard" disable fill mode for the entire application.

Idle thought:  What about setting `polygonOffset(0, 0)` for layers that don't need it (or have falsy values).  But that assumes that `polygonOffset(0,0)` is equivalent to disabling it, and that the perf penalty of using `polygonOffset` is small. So perhaps safer this way.

Alternatively, we can land this PR and update when state management becomes available in luma.gl, as that is the complete solution to the state overwriting issue.